File: backend/app/schemas/celery_beat.py
"""
任务调度管理（Celery Beat 可视化管理）Pydantic Schema

调度值统一为单一表达式 schedule_expr：
- interval 类型：正整数秒数字符串（如 "60"）
- cron 类型：5 段标准 cron 表达式（如 "0 2 * * *"，顺序 分 时 日 月 周）
"""
from typing import Optional, List, Dict, Any
from datetime import datetime
from pydantic import BaseModel, Field


class BeatTaskCreate(BaseModel):
    name: str = Field(..., max_length=200)
    task_key: str = Field(..., max_length=100)
    task: str = Field(..., max_length=255)
    schedule_type: str = "interval"
    schedule_expr: str = Field(..., max_length=200)
    queue: str = "default"
    args: Optional[List[Any]] = []
    kwargs: Optional[Dict[str, Any]] = {}
    description: Optional[str] = ""
    enabled: bool = True


class BeatTaskUpdate(BaseModel):
    id: int
    name: Optional[str] = None
    task_key: Optional[str] = None
    task: Optional[str] = None
    schedule_type: Optional[str] = None
    schedule_expr: Optional[str] = None
    queue: Optional[str] = None
    args: Optional[List[Any]] = None
    kwargs: Optional[Dict[str, Any]] = None
    description: Optional[str] = None
    enabled: Optional[bool] = None


class BeatTaskStatus(BaseModel):
    id: int
    enabled: bool


class BeatTaskRunOnce(BaseModel):
    id: int


class BeatTaskResponse(BaseModel):
    id: int
    name: str
    task_key: Optional[str] = None
    task: str
    schedule_type: str
    schedule_expr: str = ""
    queue: str
    args: Optional[List[Any]] = []
    kwargs: Optional[Dict[str, Any]] = {}
    description: Optional[str] = ""
    enabled: bool = True
    last_run_at: Optional[datetime] = None
    total_run_count: int = 0
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None


class BeatTaskListResponse(BaseModel):
    total: int
    page: int
    page_size: int
    items: List[BeatTaskResponse]


class BeatTaskLogResponse(BaseModel):
    id: int
    task_name: str
    task_key: Optional[str] = None
    task_id: str
    args: Optional[List[Any]] = []
    kwargs: Optional[Dict[str, Any]] = {}
    queue: Optional[str] = "default"
    state: str
    started_at: Optional[datetime] = None
    finished_at: Optional[datetime] = None
    duration_ms: Optional[int] = None
    exception: Optional[str] = None
    traceback: Optional[str] = None
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class BeatTaskLogListResponse(BaseModel):
    total: int
    page: int
    page_size: int
    items: List[BeatTaskLogResponse]

File: aits_hub_web/src/api/celeryBeat.ts
import request from './request'

export interface BeatTask {
  id: number
  name: string
  task_key?: string
  task: string
  schedule_type: 'interval' | 'cron'
  schedule_expr: string
  queue: string
  args?: any[]
  kwargs?: Record<string, any>
  description?: string
  enabled: boolean
  last_run_at?: string | null
  total_run_count: number
  created_at?: string
  updated_at?: string
}

export interface BeatTaskListResponse {
  total: number
  page: number
  page_size: number
  items: BeatTask[]
}

export interface BeatTaskLog {
  id: number
  task_name: string
  task_key?: string
  task_id: string
  args?: any[]
  kwargs?: Record<string, any>
  queue?: string
  state: 'RUNNING' | 'SUCCESS' | 'FAILURE' | 'TIMEOUT'
  started_at?: string | null
  finished_at?: string | null
  duration_ms?: number | null
  exception?: string | null
  traceback?: string | null
  created_at?: string
}

export interface BeatTaskLogListResponse {
  total: number
  page: number
  page_size: number
  items: BeatTaskLog[]
}

export function getBeatTasks(params?: { page?: number; page_size?: number; keyword?: string }) {
  return request.get<BeatTaskListResponse>('/celery/beat/list', { params })
}

export function createBeatTask(data: Partial<BeatTask>) {
  return request.post<BeatTask>('/celery/beat/create', data)
}

export function updateBeatTask(data: Partial<BeatTask>) {
  return request.put<BeatTask>('/celery/beat/update', data)
}

export function deleteBeatTask(id: number) {
  return request.delete<{ message: string }>(`/celery/beat/delete?id=${id}`)
}

export function setBeatTaskStatus(id: number, enabled: boolean) {
  return request.patch<BeatTask>('/celery/beat/status', { id, enabled })
}

export function runBeatTaskOnce(id: number) {
  return request.post<{ message: string; task_id: string }>('/celery/beat/run-once', { id })
}

export function getBeatTaskLogs(params?: {
  page?: number
  page_size?: number
  task_name?: string
  state?: string
  start_time?: string
  end_time?: string
}) {
  return request.get<BeatTaskLogListResponse>('/celery/beat/logs', { params })
}

export function getBeatTaskLogDetail(taskId: string) {
  return request.get<BeatTaskLog>(`/celery/beat/log-detail/${taskId}`)
}
